scipy/stats/tests/test_mstats_basic.py

```diff
@@ -1124,6 +1124,32 @@ def test_zero_division(self):
             assert_array_equal(t, np.array([np.nan, np.nan]))
             assert_array_equal(p, np.array([np.nan, np.nan]))
 
+    def test_bad_alternative(self):
+        msg = r"alternative must be 'less', 'greater' or 'two-sided'"
+        with pytest.raises(ValueError, match=msg):
+            mstats.ttest_ind([1, 2, 3], [4, 5, 6], alternative='foo')
+
+    @pytest.mark.parametrize("alternative", ["less", "greater"])
+    def test_alternative(self, alternative):
+        x = stats.norm.rvs(loc=10, scale=2, size=100, random_state=123)
+        y = stats.norm.rvs(loc=8, scale=2, size=100, random_state=123)
```

Comment:
I would suggest adjusting the way the random state is fixed. Doing it this way, every element of x is bigger than the corresponding element of y, so we get p-values of either 0 or 1. That also means there's no distinction in the p-value for `alternative="greater"` vs `alternative='two-sided'`. The test would be a little more sensitive if it had to calculate the correct decimal value, which would be distinct for each alternative.

Reply:
I changed the scale of x to 5 and decreased the sample size to 25 with seed=42. This now generates non-zero p-values which would test ttest_rel alternatives up to a few decimal points.